You can't rehire Larry from M.E.R.C when he is drunk

[//]: # (Thank you for taking the time to report an issue or propose an enhancement. Questions are also welcome.)
[//]: # (If you have more than one item to report, please open separate issues for each one!)

[//]: # (This text will not be visible and you can freely delete it. Check the Preview tab for a cleaner look.)

### What is wrong

Hello 

From M.E.R.C we can hire Larry. When he is part of our squad we can give him an alcohol and then he became Larry drunk from profile 47. 

After we release him from our game, he is returning to the M.E.R.C offer. But after this, he is a Larry drunk in offer. When we want to rehire him again, then the game crashes.

Regards

[//]: # (For bugs, please fill in this table [otherwise delete it]. The lines don't have to stay aligned.)

| JA2S version | Game version        | Vanilla bug? | Used to work? |
| ------------ | ------------------- | ------------ | ------------- |
| 0.21.0 | JA2 Gold 1.12 ENG  |    I don't know    |    I dont' know     |

[//]: # (Please also include:)
[//]: # (- how to reproduce the bug and, if possible, a save game)
[//]: # Windows 10
```
2024-01-28T16:25:06.8438798Z [INFO] game\GameLoop.cc: Version Label: Stracciatella v0.21.0-git+61938e1
2024-01-28T16:25:06.8453496Z [INFO] game\GameLoop.cc: Version #:     Build 04.12.02
2024-01-28T16:40:49.5721972Z [ERROR] game\Strategic\Queen_Command.cc: All enemy slots should be free at this point.  Only 31 of 32 are available.
			Trying to add 0 admins, 0 troops, and 0 elites.
2024-01-28T16:41:08.7122645Z [ERROR] game\Tactical\PathAI.cc: Trying to calculate path from off-world gridno 25601 to 12148
2024-01-28T16:41:08.7135536Z [ERROR] game\Tactical\PathAI.cc: Trying to calculate path from off-world gridno 25601 to 12148
2024-01-28T16:41:08.7141188Z [ERROR] game\Tactical\PathAI.cc: Trying to calculate path from off-world gridno 25601 to 12148
2024-01-28T16:41:08.7145492Z [ERROR] game\Tactical\PathAI.cc: Trying to calculate path from off-world gridno 25601 to 12148
2024-01-28T16:41:23.3777198Z [ERROR] game\Tactical\PathAI.cc: Trying to calculate path from off-world gridno 25601 to 23273
2024-01-28T16:41:23.3793935Z [ERROR] game\Tactical\PathAI.cc: Trying to calculate path from off-world gridno 25601 to 23273
2024-01-28T16:41:23.3799124Z [ERROR] game\Tactical\PathAI.cc: Trying to calculate path from off-world gridno 25601 to 23273
2024-01-28T16:41:23.3804183Z [ERROR] game\Tactical\PathAI.cc: Trying to calculate path from off-world gridno 25601 to 23273
2024-01-28T16:41:37.9568736Z [ERROR] game\Tactical\PathAI.cc: Trying to calculate path from off-world gridno 25601 to 12972
2024-01-28T16:41:37.9586957Z [ERROR] game\Tactical\PathAI.cc: Trying to calculate path from off-world gridno 25601 to 12972
2024-01-28T16:41:37.9592236Z [ERROR] game\Tactical\PathAI.cc: Trying to calculate path from off-world gridno 25601 to 12972
2024-01-28T16:41:37.9597896Z [ERROR] game\Tactical\PathAI.cc: Trying to calculate path from off-world gridno 25601 to 12972
2024-01-28T16:41:51.6536246Z [ERROR] game\Tactical\PathAI.cc: Trying to calculate path from off-world gridno 25601 to 4702
2024-01-28T16:41:51.6548425Z [ERROR] game\Tactical\PathAI.cc: Trying to calculate path from off-world gridno 25601 to 4702
2024-01-28T16:41:51.6556962Z [ERROR] game\Tactical\PathAI.cc: Trying to calculate path from off-world gridno 25601 to 4702
2024-01-28T16:41:51.6564439Z [ERROR] game\Tactical\PathAI.cc: Trying to calculate path from off-world gridno 25601 to 4702
2024-01-28T16:53:23.0827931Z [ERROR] game\Strategic\Strategic_AI.cc: Attempting to send reinforcements from a garrison that doesn't have any! -- KM:0 (with prior saved game and Debug Log)
2024-01-28T16:54:18.5667678Z [ERROR] sgp\Debug.h: Assertion failed in C:\projects\ja2-stracciatella\src\game\Tactical\Items.cc, line 2681:

2024-01-28T16:54:18.5682749Z [ERROR] sgp\Debug.h: Assertion failed in C:\projects\ja2-stracciatella\src\game\Tactical\Items.cc, line 2681:
```

--- a/src/game/Tactical/Soldier_Profile.cc
+++ b/src/game/Tactical/Soldier_Profile.cc
@@ -990,7 +990,7 @@ SOLDIERTYPE* SwapLarrysProfiles(SOLDIERTYPE* const s)
 */
 
 	memcpy(dst.bInvStatus, src.bInvStatus, sizeof(dst.bInvStatus));
-	memcpy(dst.bInvNumber, src.bInvStatus, sizeof(dst.bInvNumber));
+	memcpy(dst.bInvNumber, src.bInvNumber, sizeof(dst.bInvNumber));
 	memcpy(dst.inv,        src.inv,        sizeof(dst.inv));
 
 	DeleteSoldierFace(s);

